x-pack/plugins/notifications/server/plugin.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { PluginInitializerContext, CoreSetup, CoreStart, Plugin } from '@kbn/core/server';
+import type {
+  NotificationsPluginSetupDeps,
+  NotificationsPluginStartDeps,
+  NotificationsPluginStart,
+} from './types';
+import type { NotificationsConfigType } from './config';
+import { EmailServiceProvider } from './services/connectors_email_service_provider';
+
+export class NotificationsPlugin implements Plugin<void, NotificationsPluginStart> {
+  private emailServiceProvider: EmailServiceProvider;
```

Comment:
In order to minimize the plugin's logic, a provider has been defined, which also has the `setup()` and `start()` lifecycle hooks.

This way, all the service startup checks and initializations are abstracted away from the plugin's code.